tahoe_gui/tahoe.py

```diff
@@ -15,48 +15,39 @@
     os.environ["PATH"] += os.pathsep + os.path.join(
         os.path.dirname(sys.executable), 'Tahoe-LAFS')
 
+class TahoeEnvironment:
+    """
+    I represent a virtualenv that has Tahoe-LAFS installed in it.
+    """
+    def __init__(self, basedir):
+        self._base = basedir
 
-class CommandProtocol(ProcessProtocol):
-    def __init__(self, parent, callback_trigger=None):
-        self.parent = parent
-        self.trigger = callback_trigger
-        self.done = Deferred()
-
-    def outReceived(self, data):
-        for line in data.decode('utf-8').strip().split('\n'):
-            self.parent.out_received(line)
-            if not self.done.called and self.trigger and self.trigger in line:
-                self.done.callback(None)
-
-    def errReceived(self, data):
-        for line in data.decode('utf-8').strip().split('\n'):
-            self.parent.err_received(line)
-
-    def processEnded(self, reason):
-        if not self.done.called:
-            self.done.callback(None)
-
-    def processExited(self, reason):
-        if not self.done.called and not isinstance(reason.value, ProcessDone):
-            self.done.errback(reason)
+    @inlineCallbacks
+    def create(self):
+        """
+        :return: a Deferred that fires when we've installed a virtualenv
+        """
 
 
-class Tahoe():
-    def __init__(self, nodedir=None):
-        self.nodedir = nodedir
+class TahoeClient:
+    def __init__(self, nodedir):
```

Comment:
No, pip alone on a fresh Debian is not sufficient. Okay, lets scratch the virtualenv entirely.

I guess the reason I was imagining managing a virtualenv is because the OS packages take so long to update, typically. But, perhaps that's just the way it is etc and we should live with that.

(i.e. to play my own devil's advocate: instead of managing a virtualenv, the best next-step would be for leastauthority to manage a `deb.leastauthority.com` or similar and push "updated faster" packages there)